modules/sdk-core/src/bitgo/tss/ecdsa/ecdsa.ts

```diff
@@ -0,0 +1,120 @@
+import { ECDSA, Ecdsa } from './../../../account-lib/mpc/tss';
+import { DecryptableNShare, CombinedKey, SigningMaterial, EncryptedNShare } from './types';
+import { encryptAndSignText, readSignedMessage } from './../../utils';
+
+type NShare = ECDSA.NShare;
+type KeyShare = ECDSA.KeyShare;
+/**
+ * Combines NShares to combine the final TSS key
+ * This can only be used to create the User or Backup key since it requires the common keychain from BitGo first
+ *
+ * @param params.keyShare - TSS key share
+ * @param params.encryptedNShares - encrypted NShares with information on how to decrypt
+ * @param params.commonKeychain - expected common keychain of the combined key
+ * @returns {CombinedKey} combined TSS key
+ */
+export async function createCombinedKey(params: {
+  keyShare: KeyShare;
+  encryptedNShares: DecryptableNShare[];
+  commonKeychain: string;
+}): Promise<CombinedKey> {
+  const MPC = new Ecdsa();
+
+  const { keyShare, encryptedNShares, commonKeychain } = params;
+  const nShares: NShare[] = [];
+
+  let bitgoNShare: NShare | undefined;
```

Comment:
Can we avoid using `| undefined` ?

Reply:
Typescript will be not happy. it's possible that the caller of this method did not provide bitgoNShare. There's check on line 67 to check for bitgoNShare.